Description: amd64_fixes.patch
Author: Nick Bane <nick@enomem.co.uk>

---

Index: trunk/src/numutils.c
===================================================================
--- trunk.orig/src/numutils.c	2012-09-04 15:46:38.000000000 +0100
+++ trunk/src/numutils.c	2012-09-04 15:46:47.000000000 +0100
@@ -3,27 +3,30 @@
 /* (Little endian) format to native types                        */
 /*                                                               */
 /* This file is part of catdoc project                           */
-/* (c) Victor Wagner 1996-2003, (c) Alex Ott 2003	             */
+/* (c) Victor Wagner 1996-2003, (c) Alex Ott 2003                */
 /*****************************************************************/
 
+/* FIXME - check that this is always available too NCB 8/7/12 */
+#include <stdint.h>
 
 /********************************************************************/
 /* Reads 2-byte LSB  int from buffer at given offset platfom-indepent
  * way
- *********************************************************************/ 
-unsigned int getshort(unsigned char *buffer,int offset) {
-	return (unsigned short int)buffer[offset]|((unsigned short int)buffer[offset+1]<<8);
-}  
+ *********************************************************************/
+uint32_t getshort(unsigned char *buffer,int offset) {
+	return (uint16_t)buffer[offset]|((uint16_t)buffer[offset+1]<<8);
+}
+
 /********************************************************************/
 /* Reads 4-byte LSB  int from buffer at given offset almost platfom-indepent
  * way
- *********************************************************************/ 
-long int getlong(unsigned char *buffer,int offset) {
-	return (long)buffer[offset]|((long)buffer[offset+1]<<8L)
-		|((long)buffer[offset+2]<<16L)|((long)buffer[offset+3]<<24L);
-}  
+ *********************************************************************/
+int32_t getlong(unsigned char *buffer,int offset) {
+	return (int32_t)buffer[offset]|((int32_t)buffer[offset+1]<<8L)
+		|((int32_t)buffer[offset+2]<<16L)|((int32_t)buffer[offset+3]<<24L);
+}
 
-unsigned long int getulong(unsigned char *buffer,int offset) {
-	return (unsigned long)buffer[offset]|((unsigned long)buffer[offset+1]<<8L)
-		|((unsigned long)buffer[offset+2]<<16L)|((unsigned long)buffer[offset+3]<<24L);
-}  
+uint32_t getulong(unsigned char *buffer,int offset) {
+	return (uint32_t)buffer[offset]|((uint32_t)buffer[offset+1]<<8L)
+		|((uint32_t)buffer[offset+2]<<16L)|((uint32_t)buffer[offset+3]<<24L);
+}
Index: trunk/src/catdoc.h
===================================================================
--- trunk.orig/src/catdoc.h	2012-09-04 15:46:38.000000000 +0100
+++ trunk/src/catdoc.h	2012-09-04 15:46:47.000000000 +0100
@@ -20,6 +20,9 @@
 #include <stdio.h>
 #include <ctype.h>
 
+/* FIXME - check that this is always available too NCB 8/7/12 */
+#include <stdint.h>
+
 /* This is our own file */
 #include "ole.h"
 
@@ -230,7 +233,7 @@
 	char *strdup(const char *s);
 #endif
 /* numeric conversions */	
-long int getlong(unsigned char *buffer,int offset);
-unsigned long int getulong(unsigned char *buffer,int offset);
-unsigned int getshort(unsigned char *buffer,int offset);
+int32_t getlong(unsigned char *buffer,int offset);
+uint32_t getulong(unsigned char *buffer,int offset);
+uint16_t getshort(unsigned char *buffer,int offset);
 #endif
