Description: charset_fix.patch
Author: Nick Bane <nick@enomem.co.uk>

---

Index: trunk/src/rtfread.c
===================================================================
--- trunk.orig/src/rtfread.c	2012-08-30 12:50:31.000000000 +0100
+++ trunk/src/rtfread.c	2012-08-30 12:50:32.000000000 +0100
@@ -940,7 +940,9 @@
 	const char *charset_name;
 	char *save_buf = input_buffer;
 	if (forced_charset) return;
-	if (getCharset(group->codepage)) return;
+	group->charset = getCharset(group->codepage);
+	if (group->charset)
+		return;
 	charset_name = charset_from_codepage(group->codepage);
 	check_charset(&source_csname,charset_name);
 	input_buffer=NULL;
