--- a/configure.in
+++ b/configure.in
@@ -100,9 +100,9 @@
 dnl Checks for library functions.
 AC_FUNC_SETVBUF_REVERSED
 
-if test "$GCC" = "yes"; then
-	CFLAGS="-g -O2 -Wall"
-fi
+#if test "$GCC" = "yes"; then
+#	CFLAGS="-g -O2 -Wall"
+#fi
 
 AC_DEFINE_UNQUOTED([SOURCE_CHARSET],"$sourcecharset",[Character encoding used by default for 8-bit source files])
 AC_DEFINE_UNQUOTED([TARGET_CHARSET],"$targetcharset",[Output character encoding used by default, if impossible to determine encoding from locale])
@@ -123,6 +123,6 @@
 AC_SUBST(installroot)
 AC_SUBST(catdoc_version)
 AC_SUBST(CFLAGS)
-AC_SUBST(WORDS_BIGENDIAN DEFS)
+AC_SUBST(WORDS_BIGENDIAN)
 AC_CONFIG_HEADER(src/config.h)
 AC_OUTPUT(doc/Makefile charsets/Makefile src/Makefile Makefile doc/catdoc.1 doc/xls2csv.1 doc/wordview.1 doc/catppt.1)
--- a/src/Makefile.in
+++ b/src/Makefile.in
@@ -74,12 +74,12 @@
 #	$(CC) -c $(CFLAGS) $*
 all: @buildtargets@
 catdoc: $(OBJ)
-	$(CC)  -o catdoc  $(OBJ)
+	$(CC)  -o catdoc  $(OBJ) $(LDFLAGS)
 xls2csv: $(OBJXLS)
-	$(CC) -o xls2csv $(OBJXLS) -lm
+	$(CC) -o xls2csv $(OBJXLS) -lm $(LDFLAGS)
 
 catppt: $(OBJPPT)
-	$(CC) -o catppt $(OBJPPT) -lm
+	$(CC) -o catppt $(OBJPPT) -lm $(LDFLAGS)
 
 install: @installtargets@
 install-catdoc:catdoc xls2csv catppt 
--- a/doc/Makefile.in
+++ b/doc/Makefile.in
@@ -10,7 +10,7 @@
 exec_prefix = @exec_prefix@
 bindir = @bindir@
 libdir = @libdir@
-mandir = @man1dir@
+mandir = @mandir@
 manext = @manext@ 
 %.txt: %.1
 	nroff -mandoc $< |col -bx >$@
@@ -22,17 +22,18 @@
 
 dosdoc: catdoc.txt catdoc.ps xls2csv.txt xls2csv.ps catppt.txt catppt.ps
 clean:
-	rm build
+	$(RM) -f catppt.txt catppt.ps catppt.1 \
+            catdoc.1 catdoc.txt catdoc.ps \
+            xls2csv.1 xls2csv.txt xls2csv.ps \
+            wordview.1 wordview.txt wordview.ps build
 install: @installtargets@
 install-catdoc: catdoc.1 xls2csv.1 catppt.1
 	../mkinstalldirs $(installroot)$(mandir)
-	$(INSTALL) -m 644 catdoc.1 $(installroot)$(mandir)/catdoc$(manext)
-	$(INSTALL) -m 644 xls2csv.1 $(installroot)$(mandir)/xls2csv$(manext)
-	$(INSTALL) -m 644 catppt.1 $(installroot)$(mandir)/catppt$(manext)
+	$(INSTALL) -m 644 catdoc.1 $(installroot)$(mandir)/man1/catdoc.1
+	$(INSTALL) -m 644 xls2csv.1 $(installroot)$(mandir)/man1/xls2csv.1
+	$(INSTALL) -m 644 catppt.1 $(installroot)$(mandir)/man1/catppt.1
 install-wordview:	
 	../mkinstalldirs $(installroot)$(mandir)
-	$(INSTALL) -m 644 wordview.1 $(installroot)$(mandir)/wordview$(manext)
-# Following rules are primarily for developers use. I doubt that you
-# need to rebuild these versions of documentation
-distclean: catppt.txt catppt.ps catdoc.1 catdoc.txt catdoc.ps xls2csv.1  xls2csv.txt xls2csv.ps
+	$(INSTALL) -m 644 wordview.1 $(installroot)$(mandir)/man1/wordview.1
+distclean: clean
 	rm Makefile
